### ISSUE TITLE
Explicitly define domain of Pyomo Params

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/calliope#309]
## Problem description

There are loads of Pyomo warnings in the logs of my model runs:

```
WARNING: DEPRECATED: The default domain for Param objects is 'Any'.  However,
    we will be changing that default to 'Reals' in the future.  If you really
    intend the domain of this Param (resource_unit) to be 'Any', you can
    suppress this warning by explicitly specifying 'within=Any' to the Param
    constructor.  (deprecated in 5.6.9, will be removed in 6.0)
```

I assume it's because (some?) Params in Calliope do not define a domain explicitly.

## Calliope version

0.6.5

## Pyomo version

5.6.9

**Comment by brynpickering** [comment]
Yeah, we don't define the domain for any Params as far as I'm aware. We could probably just make it an easy fix with 'within=Any', but a smarter way would be to match the dtypes in the xarray dataset: 'Object' -> 'Any', 'bool' -> Binary, 'int/float' -> Reals. Could even go one step further and differentiate between `NonNeg` and `Neg` Reals.

The advantage of the smarter solution would be that on updating a Pyomo model to rerun it, we could remove the warning about us having no idea what's going on, since (I expect) Pyomo would simply complain if the dtype of the new data is incorrect.

**Comment by brynpickering** [comment]
Fixed in #313
